### PR TITLE
Accounts RPC Handler

### PR DESCRIPTION
**Pull request by natenichols**
Adds handlers for Account RPCs

`account_lines` `account_channels`, `account_objects`, `account_currencies`, `account_offers`

---
This change is [Reviewable](https://reviewable.io/reviews/cjcobb23/xrpl-reporting/1)

### CODE REVIEW

**Comment by natenichols** [comment]
Ignore any book_offers changes, I'll take care of that in the book offers pr

File: reporting/CassandraBackend.cpp [hunk 1]
```diff
@@ -652,8 +652,9 @@ void
 writeBook2(WriteBookCallbackData& cb)
```
Comment (cjcobb23) [inline reporting/CassandraBackend.cpp:652]: While you're in here, can we just change all of these things that are named "books2" to just "books"? And remove the old books stuff.

**Comment by cjcobb23** [comment]
> Ignore any book_offers changes, I'll take care of that in the book offers pr

Sorry I didn't see this. Ignore my comments about book_offers then

File: reporting/CassandraBackend.h [hunk 2]
```diff
@@ -631,6 +659,9 @@ class CassandraBackend : public BackendInterface
     std::optional<boost::asio::io_context::work> work_;
     std::thread ioThread_;
 
+    // std::thread indexer_;
+    uint32_t indexerShift_ = 16;
+
```
[the remark this answers is not in the document]
Comment (natenichols) [reply reporting/CassandraBackend.h]: Nope, removed

File: reporting/BackendInterface.h [hunk 3]
```diff
@@ -83,6 +83,9 @@ class BackendIndexer
     void
     deleteKey(ripple::uint256 const& key);
 
+    std::vector<ripple::uint256>
+    getCurrentOffers(ripple::uint256 const& book);
```
[the remark this answers is not in the document]
Comment (natenichols) [reply reporting/BackendInterface.h]: removed

File: reporting/CassandraBackend.h [hunk 4]
```diff
@@ -483,6 +483,34 @@ class CassandraResult
         return {first, second};
     }
 
+    std::pair<Blob, Blob>
+    getBytesTuple()
```
[the remark this answers is not in the document]
Comment (natenichols) [reply reporting/CassandraBackend.h]: Will take care of it in the book_offers pr

File: reporting/CassandraBackend.cpp [hunk 5]
```diff
@@ -1320,86 +1321,75 @@ CassandraBackend::open(bool readOnly)
...
-        if (!executeSimpleStatement(query.str()))
-            continue;
-        query = {};
+
+        query.str("");
         query << "CREATE TABLE IF NOT EXISTS " << tablePrefix << "books2"
-              << " ( book blob, sequence bigint, key blob, PRIMARY KEY "
-                 "((book, sequence), key)) WITH CLUSTERING ORDER BY (key "
+              << " ( book blob, sequence bigint, quality_key tuple<blob, blob>, PRIMARY KEY "
+                 "((book, sequence), quality_key)) WITH CLUSTERING ORDER BY (quality_key "
```
[the remark this answers is not in the document]
Comment (natenichols) [reply reporting/CassandraBackend.cpp:1400]: Will do this in book_offers

File: reporting/CassandraBackend.cpp (hunk 1 above)
Comment (natenichols) [reply reporting/CassandraBackend.cpp]: Same here

**Review by cjcobb23 — APPROVED** [review]
LGTM

**Pull request merged**